### DISCUSSION TITLE
Is it possible to add SSM Automation as Target to an Event Rule with CDK?

### DISCUSSION THREAD

**Discussion by stoennies** [aws/aws-cdk#23418] [Q&A]
Hi all,

is it possible to add SSM Automation as Target to an Event Rule with CDK?

**Comment by peterwoodworth** [comment]
It doesn't appear to me that this is supported in [CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-events-rule-target.html). You might want to open an issue in the CloudFormation coverage roadmap for this, as the [docs](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-targets.html) only mention this is for the console as far as I can tell

**Comment by andyevers** [comment]
Try using your own bind method for `rule.addTarget`. This worked for me.
```TS
// cdk imports
import { Rule } from 'aws-cdk-lib/aws-events'
import { CfnDocument } from 'aws-cdk-lib/aws-ssm'

// method content
const automation = new CfnDocument(stack, 'MyAutomation', {
    name: 'MyAutomation',
    documentType: 'Automation',
    updateMethod: 'NewVersion',
    content: {}, // your automation content here
})

const rule = new Rule(stack, 'MyRule', {
    ruleName: 'MyRule',
    eventPattern: {}, // your event pattern here
})

const accountId = '123456789012' // replace with your account id
const region = 'us-east-1' // replace with your region
rule.addTarget({
    bind: () => ({
        arn: `arn:aws:ssm:${region}:${accountId}:automation-definition/${automation.name}`,
        targetResource: automation,
        // your rule props here (role, input, etc...)
    }),
})
```

**Reply by mguja-fcc** [reply]
how would I add input transformation ?

**Reply by MyNameIsTakenOMG** [reply]
> how would I add input transformation ?

By the time I'm writing, the cdk version 2.178.1 still does not support input transformer
